pkg/server/router/credential_test.go

```diff
@@ -41,7 +41,8 @@ func TestCredentialRouter(t *testing.T) {
 
 	t.Run("Credential Service Test", func(tt *testing.T) {
 
-		bolt, err := storage.NewBoltDB()
+		bolt := &storage.BoltDB{}
```

Comment:
why not `storage.NewBoltDB()`?

Reply:
For dynamic register，implementations should load to `avaliableStorages` with `init()` in  implementation file `xxx_storage.go`.  this `xxx_storage.go` can be a part of `ssi-service` or a 3rd party repo which use `ssi-service` as a library.
if we use functions like `NewBoltDB `, different implementations will use different `NewXXXX` function so we can not create and set up implementations in `NewStorage` automaticlly (i also add an uniform standard `Init(interface{})` into ServiceStorage, so we can do `set up` with params for implementation ).

Reply:
in general I find side-effects like "init" dangerous because their behaviors are not immediately clear, and it can be confusing to debug. if we can maintain the behavior of init but make it explicit that would be preferable.

I think we can create an interface `NewStorage` which `NewBoltDB` and others would comply with to mitigate your concern